common/src/main/java/com/microsoft/identity/common/adal/internal/tokensharing/TokenShareUtility.java

```diff
@@ -58,22 +58,53 @@
 import static com.microsoft.identity.common.adal.internal.AuthenticationConstants.OAuth2.ID_TOKEN_OBJECT_ID;
 import static com.microsoft.identity.common.adal.internal.tokensharing.ITokenShareResultInternal.TokenShareExportFormatInternal.RAW;
 import static com.microsoft.identity.common.adal.internal.tokensharing.ITokenShareResultInternal.TokenShareExportFormatInternal.SSO_STATE_SERIALIZER_BLOB;
-import static com.microsoft.identity.common.java.exception.ClientException.TOKEN_CACHE_ITEM_NOT_FOUND;
 import static com.microsoft.identity.common.internal.migration.AdalMigrationAdapter.loadCloudDiscoveryMetadata;
 import static com.microsoft.identity.common.internal.migration.TokenCacheItemMigrationAdapter.renewToken;
 import static com.microsoft.identity.common.internal.migration.TokenCacheItemMigrationAdapter.sBackgroundExecutor;
+import static com.microsoft.identity.common.java.exception.ClientException.TOKEN_CACHE_ITEM_NOT_FOUND;
 
 public class TokenShareUtility implements ITokenShareInternal {
 
     private static final String TAG = TokenShareUtility.class.getSimpleName();
     private static final Map<String, String> sClaimRemapper = new HashMap<>();
     private static final String CONSUMERS_ENDPOINT = "https://login.microsoftonline.com/consumers";
+    private static final String ENVIRONMENT_GLOBAL = "login.microsoftonline.com";
+    private static final String ENVIRONMENT_GALLATIN = "login.partner.microsoftonline.cn";
+    private static final String ENVIRONMENT_ITAR = "login.microsoftonline.us";
+    private static final String ENVIRONMENT_BLACKFOREST = "login.microsoftonline.de";
+
+    private enum Environment {
+        WORLDWIDE("https://login.microsoftonline.com/common"),
+        GALLATIN("https://login.partner.microsoftonline.cn/common"),
+        BLACKFOREST("https://login.microsoftonline.de/common"),
+        ITAR("https://login.microsoftonline.us/common");
+
+        private String mCommonEndpoint;
+
+        Environment(final String commonEndpoint) {
+            mCommonEndpoint = commonEndpoint;
+        }
+
+        @Nullable
+        static Environment toEnvironment(@NonNull final String envString) {
+            switch (envString) {
+                case "login.windows.net":
+                case "login.microsoft.com":
+                case "sts.windows.net":
+                    return Environment.WORLDWIDE;
+                case "login.chinacloudapi.cn":
+                    return Environment.GALLATIN;
+                case "login.usgovcloudapi.net":
+                    return Environment.ITAR;
+                default:
+                    return null;
```

Comment:
Do we want a null return here, or should we fail some other way?

Reply:
I am thinking this should throw a client exception, log a message, and not return the token to the caller if we can't identify the authority. Will push this in another commit

Reply:
47d045ff5a33ac16755b181092a8060cd94610ac